hmrc-mongo-metrix-play-27/src/main/scala/uk/gov/hmrc/mongo/metrix/MetricOrchestrator.scala

```diff
@@ -79,20 +79,25 @@ class MetricOrchestrator(
 
   private def updateMetricRepository(
     resetToZeroFor: Option[PersistedMetric => Boolean]
-  )(implicit ec: ExecutionContext): Future[Map[String, Int]] =
+  )(implicit
+    ec: ExecutionContext
+  ): Future[Map[String, Int]] =
     for {
-      mapFromReset <- resetToZeroFor match {
-                       case Some(reset) =>
-                         metricRepository
-                           .findAll()
-                           .map(_.filter(reset).map { case PersistedMetric(name, _) => name -> 0 }.toMap)
-                       case None => Future(Map.empty[String, Int])
-                     }
-      mapFromSources <- Future.traverse(metricSources)(_.metrics)
-      mapToPersist = (mapFromReset :: mapFromSources).reduce(_ ++ _)
-      metricsToPersist = mapToPersist.map { case (name: String, value: Int) => PersistedMetric(name, value) }.toList
-      _ <- Future.traverse(metricsToPersist)(metricRepository.persist)
-    } yield mapToPersist
+      peristedMetrics   <- metricRepository.findAll
```

Comment:
typo in peristedMetrics

findAll missing parentheses.

Reply:
I think we only need the names of `peristedMetrics`. If we convert to a Set[String] here, it will simplify the code slightly.